src/test/java/com/mongodb/hibernate/jdbc/MongoStatementIntegrationTests.java

```diff
@@ -0,0 +1,349 @@
+/*
+ * Copyright 2024-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.bson.BsonDocument;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class MongoStatementIntegrationTests {
+
+    private static SessionFactory sessionFactory;
```

Comment:
I can see the IDE complaining `@NullMarked fields must be initialized` here, but not in `com.mongodb.hibernate.SessionTests`. That's because we annotated the `com.mongodb.hibernate.jdbc` package with `@NullMarked`, but did not do the same for the `com.mongodb.hibernate` package.

Let's annotate the `com.mongodb.hibernate` package with `@NullMarked`.

Reply:
when I added `@NullMarked` to `com.mongodb.hibernate`, IDE complains about both classes. Not sure how to solve the issue.
I think the workaround is to add `@Nullable` to these variables just treating them as normal non-test classes.

Reply:
> IDE complains about both classes

That's as expected. The fact that it did not complain about `com.mongodb.hibernate.SessionTests` got me into the realization that the `com.mongodb.hibernate.jdbc` package wasn't annotated, despite having to be.

Reply:
I fixed both after adding the `@NullMarked` annotation to `com.mongodb.hibernate`. Note that given the integrationTest PR got merged into main, the `StatementIntegrationTest` has moved to `src/integrationTest` folder. The original `src/test` is reserved for pure unit testing cases which doesn't require external mongodb service deployment.